### PR TITLE
Add option to enable enhanced security setting in DSC pull server

### PR DESCRIPTION
**Pull request by zhenl-microsoft**
This change is [Reviewable](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/223)

### CODE REVIEW

**Comment by johlju** [comment]
Suggestion: The SChannel.psm1 module, shouldn't we make that a separate resource? For example, setting the cipher suites order seems it could be used for other scenarios too.

**Comment by zhenl-microsoft** [comment]
@johlju Thanks for the feedback. This is a very good suggestion. As we foresee that finalize the design and implement a new resource SChannel could take much longer time, the fix we have here is to unblock the request of setting up pull server with enhanced security. Feel free to open an issue and even work on contributing a new SChannel resource. Once we do have a good working resource, we could update this resource to use it.

**Comment by kwirkykat** [comment]
@zhenl-microsoft Please [update the README](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#update-the-release-notes) with these changes.

---

Reviewed 2 of 4 files at r1, 4 of 4 files at r2.
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xDSCWebService/SChannel.psm1, line 62 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223#-KRFoJsYhotpHsnvwF8S:-KRFoJsYhotpHsnvwF8T:b-iwixen) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/48ac1fe35da93969703825098a1b6ee9f3b14169/DSCResources/MSFT_xDSCWebService/SChannel.psm1#L62)):*
> ```PowerShell
> # ===========================================================
> 
> function Test-Protocol
> ```

Can you please include help comments above each of the new functions in this file?
Since they don't have parameters all you need is:
```
<#
    .SYNOPSIS
        <Insert function summary here>
#>
function ...
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223#-:-KRFow1gi3wxOoBzSjzm:blqrftq)*

**Comment by zhenl-microsoft** [comment]
Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xDSCWebService/SChannel.psm1, line 62 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223#-KRFoJsYhotpHsnvwF8S:-KRFvhsqeuphHfN-FGTC:b-896fix) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/48ac1fe35da93969703825098a1b6ee9f3b14169/DSCResources/MSFT_xDSCWebService/SChannel.psm1#L62)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Can you please include help comments above each of the new functions in this file?
> Since they don't have parameters all you need is:
> ```
> <#
>     .SYNOPSIS
>         <Insert function summary here>
> #>
> function ...
> ```
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223)*

**Comment by zhenl-microsoft** [comment]
Done.

---

Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223#-:-KRFw-EJJXBhlOPv5MFq:b-896fix)*

**Comment by kwirkykat** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/223#-:-KRFx4r84xZRfpnDq6TG:bnfp4nl)*

File: DSCResources/MSFT_xDSCWebService/SChannel.psm1 [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+#
+
+# ============ Up to Date Security Settings Block =========
+$insecureProtocols            = @("SSL 2.0", "SSL 3.0", "TLS 1.0", "PCT 1.0", "Multi-Protocol Unified Hello")
+$secureProtocols              = @("TLS 1.1", "TLS 1.2")
+
+$insecureCiphers              = @('DES 56/56', 'NULL', 'RC2 128/128', 'RC2 40/128', 'RC2 56/128', 'RC4 128/128', 'RC4 40/128', 'RC4 56/128', 'RC4 64/128', 'RC4 128/128')
+$secureCiphers                = @('AES 128/128','AES 256/256','Triple DES 168/168')
+
+$enableHashes                 = @("SHA","SHA256","SHA384","SHA512")
```
Comment (megamorf) [inline DSCResources/MSFT_xDSCWebService/SChannel.psm1]: SHA1 isn't considered state of the art anymore. I'd be good to remove it from the list.

**Comment by kwirkykat** [comment]
@megamorf Can you open this as an issue?

**Pull request merged**